frontend/pages/SoftwarePage/SoftwarePage.tsx

```diff
@@ -256,23 +258,43 @@ const SoftwarePage = ({ children, router, location }: ISoftwarePageProps) => {
     }
   };
 
+  // NOTE: used to reset page number to 0 when modifying filters
+  // NOTE: Solution reused from ManageHostPage.tsx
+  useEffect(() => {
+    setResetPageIndex(false);
+    console.log("queryParams.page", queryParams.page);
+    console.log("page.toString()", page.toString());
+
+    // Ensures setting page to same page as seen in URL
+    if (initialPage !== page) {
+      setPage(initialPage);
+    }
+  }, [queryParams, page]);
```

Comment:
The table container logic is awful because it wants to be the source of truth for pages, but the URL is the source of truth, so there's some funky forcing of page numbers from the pagination of the table to the URL and from the URL to the table. Def update of the table/router libraries should look into better solutions.